expr/fieldexpr.go

```diff
@@ -90,7 +90,10 @@ func FieldExprToString(e Evaluator) string {
 		rhs := FieldExprToString(e.field)
 		return lhs + rhs
 	case *Literal:
-		return e.zv.String()
+		if e.zv.Type != zng.TypeString {
+			panic("expr: unexpected type " + e.zv.Type.String())
```

Comment:
This changed in the dot-record merge and has an error return.  I don't think panic should be called here because then the caller needs to make sure no weird stuff crept through.  Better to get an error (which you will see when you rebase though your check isn;t there)

Reply:
The *Literal case is no longer needed, so it goes away in #1487.